### PR TITLE
test: cover index account history stage with stage_test_suite_ext tests

### PR DESCRIPTION
**Pull request by int88**
fixes part of #3232

### CODE REVIEW

**Comment by int88** [comment]
@shekhirin PTAL :)

File: crates/stages/src/stages/index_account_history.rs [hunk 1]
```diff
@@ -357,4 +362,90 @@ mod tests {
...
+
+        fn stage(&self) -> Self::S {
+            Self::S { commit_threshold: self.commit_threshold }
+        }
+    }
+
+    impl ExecuteStageTestRunner for IndexAccountHistoryTestRunner {
+        type Seed = ();
+
+        fn seed_execution(&mut self, input: ExecInput) -> Result<Self::Seed, TestRunnerError> {
```
Comment (shekhirin) [inline crates/stages/src/stages/index_account_history.rs:392]: I think what we want here is to seed the database with some random blocks and account changesets, as we do in e.g. Merkle stage: https://github.com/paradigmxyz/reth/blob/bb57556a6ab84f1c9f09eebd1f618220430b5101/crates/stages/src/stages/merkle.rs#L472

Then, in `validate_execution` we should dynamically calculate the shards from database data, and compare it with the contents of `AccountHistory` table.

**Review by shekhirin — CHANGES_REQUESTED** [review]

File: crates/stages/src/stages/index_account_history.rs (hunk 1 above)
Comment (int88) [reply crates/stages/src/stages/index_account_history.rs:399]: @shekhirin updated! PTAL

**Comment by onbjerg** [comment]
Ping @shekhirin

**Review by shekhirin — APPROVED** [review]
LGTM, small nit. Sorry for the slow review 😑

Would also like to get @rakita review on this tests since he wrote the stage implementation

File: crates/stages/src/stages/index_account_history.rs [hunk 2]
```diff
@@ -357,4 +369,136 @@ mod tests {
...
+                    // chunk indices and insert them in shards of N size.
+                    let mut chunks = indices
+                        .iter()
+                        .chunks(sharded_key::NUM_OF_INDICES_IN_SHARD)
+                        .into_iter()
+                        .map(|chunks| chunks.map(|i| *i as usize).collect::<Vec<usize>>())
+                        .collect::<Vec<_>>();
+                    let last_chunk = chunks.pop();
+
+                    chunks.into_iter().try_for_each(|list| -> Result<_, TestRunnerError> {
```
[the remark this answers is not in the document]
Comment (int88) [reply crates/stages/src/stages/index_account_history.rs]: @shekhirin updated!

**Comment by int88** [comment]
friendly ping @rakita

**Review by rakita — APPROVED** [review]
one nit, but otherwise it lgtm!

Thank you for the ping and I apologize for missing this I was on vacation week before.

File: crates/stages/src/stages/index_account_history.rs [hunk 3]
```diff
@@ -75,7 +75,19 @@ mod tests {
...
+    };
+    use itertools::Itertools;
+    use reth_db::{cursor::DbCursorRO, models::sharded_key, transaction::DbTx};
+    use reth_interfaces::test_utils::{
+        generators,
+        generators::{random_block_range, random_contract_account_range, random_transition_range},
+    };
+    use reth_primitives::{Address, BlockNumber, H256};
+
+    stage_test_suite_ext!(IndexAccountHistoryTestRunner, index_account_history);
```
[the remark this answers is not in the document]
Comment (int88) [reply crates/stages/src/stages/index_account_history.rs]: @rakita updated! :)

**Pull request merged**